p2p/pex/pex_reactor.go

```diff
@@ -339,19 +342,10 @@ func (r *PEXReactor) ensurePeers() {
 	}
 
 	// Dial picked addresses
-	for _, item := range toDial {
+	for _, addr := range toDial {
 		go func(picked *p2p.NetAddress) {
-			err := r.Switch.DialPeerWithAddress(picked, false)
-			if err != nil {
-				r.Logger.Error("Dialing failed", "err", err)
-				// TODO: detect more "bad peer" scenarios
-				if _, ok := err.(p2p.ErrSwitchAuthenticationFailure); ok {
-					r.book.MarkBad(picked)
-				} else {
-					r.book.MarkAttempt(picked)
-				}
-			}
-		}(item)
+			r.dialPeer(picked)
+		}(addr)
```

Comment:
Can't we call `go r.dialPeer(addr)` directly?

Reply:
https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables ?

Reply:
Here `addr` is passed to `dialPeer` and evaluate at each iteration and placed on the stack of the routine, similar to how you pass it to the clouse at the moment.

Reply:
okay.. thanks